test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -0,0 +1,232 @@
+    if (this.getAttribute("privacy")) {
+      div1.setAttribute("data-dd-privacy", this.getAttribute("privacy"));
+      div2.setAttribute("data-dd-privacy", this.getAttribute("privacy"));
+    }
+    const label = document.createElement("label");
+    label.setAttribute("for", "input-" + this.getAttribute("id"));
+    label.setAttribute("id", "label-" + this.getAttribute("id"));
+    label.innerText = "field " + this.getAttribute("id") + ": ";
+    const input = document.createElement("input");
+    input.setAttribute("id", "input-" + this.getAttribute("id"));
+    input.setAttribute("value", "toto");
+    input.addEventListener('input', (e) => {
+      p.textContent = e.target.value
+    })
+    div1.appendChild(label);
+    div1.appendChild(input);
+    div1.setAttribute("id", "div1-" + this.getAttribute("id"));
+    this.shadowRoot.appendChild(div1);
+    this.shadowRoot.appendChild(div2);
+  }
+}
+window.customElements.define("my-open-web-component", MyOpenWebComponent);
+</script>
+`
+
+describe('recorder with shadow DOM', () => {
+  describe('full snapshot', () => {
+    createTest('can overwrite with mask-user-input')
+      .withRum({ defaultPrivacyLevel: 'allow', enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <div id="wrapper-outside" data-dd-privacy="mask-user-input"><my-open-web-component id="outside" /></div>
+          <div id="wrapper-inside"><my-open-web-component privacy="mask-user-input" id="inside" /></div>
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await flushEvents()
+
+        expect(serverEvents.sessionReplay.length).toBe(1)
+
+        const fullSnapshot = findFullSnapshot(getFirstSegment(serverEvents))!
+        expect(fullSnapshot).toBeTruthy()
+
+        const {
+          input: outsideInput,
+          shadowHost: outsideShadowHost,
+          textContent: outsideTextContent,
+        } = findElementsInShadowDom(fullSnapshot.data.node, 'outside')
+        expect(outsideShadowHost?.isShadowHost).toBeTrue()
+        expect(outsideInput?.attributes.value).toBe('***')
+        expect(outsideTextContent).toBe('toto')
+
+        const {
+          input: insideInput,
+          shadowHost: insideShadowHost,
+          textContent: insideTextContent,
+        } = findElementsInShadowDom(fullSnapshot.data.node, 'inside')
+        expect(insideShadowHost?.isShadowHost).toBeTrue()
+        expect(insideInput?.attributes.value).toBe('***')
+        expect(insideTextContent).toBe('toto')
+      })
+  })
+  describe('incremental snapshot', () => {
+    createTest('record click')
+      .withRum({ enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** it appears that ‏the id is not relevent for this test and those below. Maybe remove it?